bitarray/_bitarray.c

```diff
@@ -87,13 +94,26 @@ static PyTypeObject Bitarraytype;
 
 #define BYTES(bits)  (((bits) == 0) ? 0 : (((bits) - 1) / 8 + 1))
 
-#define BITMASK(endian, i)  (((char) 1) << ((endian) ? (7 - (i)%8) : (i)%8))
+#define BITMASK(endian, i)  (((unsigned char) 1) << ((endian) ? (7 - (i)%8) : (i)%8))
 
 /* ------------ low level access to bits in bitarrayobject ------------- */
 
 #define GETBIT(self, i)  \
     ((self)->ob_item[(i) / 8] & BITMASK((self)->endian, i) ? 1 : 0)
 
+#if HAS_VECTORS
+/*
+ * Perform bitwise operation OP on 16 bytes of memory at a time.
+ */
+#define vector_op(A, B, OP) do {  \
+    vec __a, __b, __r;            \
+    memcpy(&__a, A, sizeof(vec)); \
```

Comment:
@eltoder Original PR is over two years old. I bought a new computer in the meantime : ).

@diamondman At least two copies need to happen on x86, from a slice of `ob_item` into `xmm*` registers and back. This is unavoidable and `memcpy` optimizations (currently) seem to handle this as well as can be expected. In terms of aligning memory. It would, as @eltoder put it earlier, probably end up being some extra mov (and shift) instructions to construct a copy of aligned memory. But the overall optimization should hopefully still give an order of magnitude speed up. Now I'm unfamiliar with non-x86 w.r.t. this, so I could be totally off on my assumptions.